<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>spectrometer_widget</class>
 <widget class="QWidget" name="spectrometer_widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1279</width>
    <height>642</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Spectrometer</string>
  </property>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1051</width>
     <height>621</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="main_layout" stretch="1,4">
    <property name="sizeConstraint">
     <enum>QLayout::SetMinimumSize</enum>
    </property>
    <property name="leftMargin">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="rightMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item>
     <widget class="QFrame" name="control_frame">
      <property name="maximumSize">
       <size>
        <width>350</width>
        <height>600</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="control_layout">
       <property name="spacing">
        <number>2</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetMaximumSize</enum>
       </property>
       <item>
        <layout class="QHBoxLayout" name="initialise_layout">
         <item>
          <widget class="QPushButton" name="initialise_button">
           <property name="text">
            <string>Initialise</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="initialise_label">
           <property name="text">
            <string>Status: not connected</string>
           </property>
           <property name="textFormat">
            <enum>Qt::AutoText</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="integration_time_layout">
         <item>
          <widget class="QLabel" name="integration_time_label">
           <property name="text">
            <string>Integration time (ms)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="integration_time_edit">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <property name="text">
            <string>100</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="scans_average_layout">
         <item>
          <widget class="QLabel" name="scans_average_label">
           <property name="text">
            <string>Scans to average</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="scans_average_edit">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <property name="text">
            <string>1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="filter_layout">
         <item>
          <widget class="QCheckBox" name="filter_checkbox">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Filter from (nm)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="filter_lower_limit_edit">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>70</width>
             <height>100</height>
            </size>
           </property>
           <property name="layoutDirection">
            <enum>Qt::RightToLeft</enum>
           </property>
           <property name="text">
            <string>400</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="to_label">
           <property name="text">
            <string>to</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="filter_upper_limit_edit">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>70</width>
             <height>100</height>
            </size>
           </property>
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="text">
            <string>1100</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="electrical_dark_layout">
         <item>
          <widget class="QCheckBox" name="electrical_dark_checkbox">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Correct for electrical dark</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="substract_background_layout">
         <item>
          <widget class="QCheckBox" name="substract_background_checkbox">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Substract background</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QVBoxLayout" name="measure_layout">
         <item>
          <widget class="QPushButton" name="single_spectrum_button">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Get Single Spectrum</string>
           </property>
           <property name="checkable">
            <bool>false</bool>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
           <property name="autoRepeat">
            <bool>false</bool>
           </property>
           <property name="autoRepeatDelay">
            <number>300</number>
           </property>
           <property name="autoDefault">
            <bool>false</bool>
           </property>
           <property name="default">
            <bool>false</bool>
           </property>
           <property name="flat">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="play_stop_layout">
           <item>
            <widget class="QPushButton" name="play_button">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Play</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="stop_button">
             <property name="text">
              <string>Stop</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="store_background_button">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Store spectrum as background</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="load_background_button">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Load spectrum as background</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="save_button">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Save</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="load_layout">
         <item>
          <widget class="QPushButton" name="load_button">
           <property name="text">
            <string>Load</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="previous_button">
           <property name="text">
            <string>Previous</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="next_button">
           <property name="text">
            <string>Next</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QGridLayout" name="plots_layout" rowstretch="1,1" columnstretch="3,2">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item row="0" column="0">
       <widget class="PlotWidget" name="current_spectrum_plot" native="true"/>
      </item>
      <item row="1" column="0">
       <widget class="PlotWidget" name="average_spectrum_plot" native="true"/>
      </item>
      <item row="0" column="1">
       <widget class="PlotWidget" name="spectrometer_counts_plot" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>100000</width>
          <height>100000</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="PlotWidget" name="spectrometer_counts_plot_2" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>100000</width>
          <height>100000</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <slots>
  <slot>custom_function()</slot>
 </slots>
</ui>
